tools/oeapkman/oeapkman

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+ARCH=$(arch)
+
+# Check if targeting OPTEE.
+optee=0
+if [ "$1" == "--optee" ]; then
+    # Optee support requires QEMU user-space emulation.
+    if ! command -v qemu-aarch64-static > /dev/null; then
+        echo "install qemu-user-static package for optee support."
+        exit 1
+    fi
+    optee=1
+    ARCH='aarch64'
+    shift
+fi
+
+# Use a recent version of proot. With older versions, apk would
+# report messages like "2 errors", that are innocuous, but catch
+# the eye.
+PROOT_URL="https://gitlab.com/proot/proot/-/jobs/981080842/\
+artifacts/download?file_type=archive"
+
+# URL to fetch alpine root fs from About 2.5 MB download.
+ALPINE_ROOTFS_URL="https://dl-cdn.alpinelinux.org/alpine/v3.14/\
```

Comment:
It would be nice to make the version configurable (even though the help could make it clear which versions are supported).

Reply:
I've added a an option `--alpine version` that allows specifying the version of Alpine Linux to use. The default is 3.14.1.
This way one can select among multiple versions of Alpine Linux.
If we introduce the notion of a tag, we can even have an instance specific for an enclave/project; but I'm holding off on it for now until clear need arises.

Next iteration I'd like to add sha256 verification of the alpine minirootfs that is downloaded. I don't see if there is a way other than maintaining a whitelist of sha256 sums. Do you have any suggestions?

Reply:
Checking against a checksum would definitely be good! Unless the images are signed, I also don't see how we can avoid keeping a list unfortunately.